calculate_states.py

```diff
@@ -0,0 +1,26 @@
+from knights_class import Knight
+
+
+def attribute_calculation(knight_warior: list[Knight]) -> list[Knight]:
+    result_list = []
+
+    for knight in knight_warior:
+        if knight.armour:
+            for detail_armour in knight.armour:
+                knight.protection += detail_armour["protection"]
+
+        knight.power += knight.weapon["power"]
+
+        if knight.potion:
+            if "power" in knight.potion["effect"]:

```

Comment:
try to avoid repeating if. you can use setattr

Reply:
Done!